Ginger/GingerCoreNET/RunLib/CLILib/CLIProcessor.cs

```diff
@@ -40,26 +42,114 @@ public void ExecuteArgs(string[] args)
             ConsoleWorkspaceEventHandler consoleWorkspaceEventHandler = new ConsoleWorkspaceEventHandler();
-            string param;
-            string value = null;
-            //if (args[0].StartsWith("ConfigFile=") || args[0].StartsWith("DynamicXML="))  // special case to support backword compatibility of old style ConfigFile=%filename%
-            if (args[0].Contains("="))
+
+            // This is the new style using verb and CommandLine Nuget to parse
+            if (args[0] == "run" || args[0] == "grid")
             {
-                string[] arg1 = args[0].Split(new[]{ '='}, 2);
-                param = arg1[0].Trim();
-                value = arg1[1].Trim();                
+                ParseArgs(args);
             }
             else
-            {                
-                param = args[0].Trim();
-                if (args.Length > 1)
+            {
+                // Support old style
+                string param;
+                string value = null;                
+                if (args[0].Contains("="))
+                {
+                    string[] arg1 = args[0].Split(new[] { '=' }, 2);
+                    param = arg1[0].Trim();
+                    value = arg1[1].Trim();
+                }
+                else
+                {
+                    param = args[0].Trim();
+                    if (args.Length > 1)
+                    {
+                        value = args[1].Trim();
+                    }
+                }
+                HandleArg(param, value);
+            }
+
+        }
+
+        private void ParseArgs(string[] args)
+        {            
+            Parser.Default.ParseArguments<RunOptions, GridOptions>(args)
+                   .WithParsed<RunOptions>(runOptions =>
+                   {
+                       // mCLIHelper.CLIType = eCLIType.Arguments;                       
+                       mCLIHandler = new CLIArgs();
+                       ExecRunOptions(runOptions);
+                       // Console.WriteLine($"Solution: {runOptions.Solution}");
+                       // Console.WriteLine($"runset: {runOptions.Runset}");
+                       // opt = runOptions;
+                   }).WithParsed<GridOptions>(gridOptions =>
+                   {
+                       StartGrid(gridOptions);                                
+                   })
+                   .WithNotParsed(errs =>
+                   {
+                       HandleCLIParseError(errs);
+                   });
+        }
+
+        private void StartGrid(GridOptions gridOptions)
+        {
+            Console.WriteLine("Starting Ginger Grid at port: " + gridOptions.Port);
+            GingerGrid gingerGrid = new GingerGrid(gridOptions.Port);   
+            gingerGrid.Start();
+        }
+
+        private void ExecRunOptions(RunOptions runOptions)
+        {
```

Comment:
i don't think we need this function it is basically duplication of PerformLoadAndExecution() method

Reply:
I fully agree, need to combine all to one unified style and remove the dup codes